### ISSUE TITLE
Ignore Empty CORS Blocks for AWS S3 Bucket

### ISSUE THREAD

**Issue by pjdufour** [hashicorp/terraform-provider-aws#7546]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Description

This PR will allow adding/removing cors blocks dynamically based on a list of allowed methods and allowed origins.  Currently, empty cors blocks cause a panic.

### New or Affected Resource(s)

* aws_s3_bucket

### Potential Terraform Configuration

```hcl
resource "aws_s3_bucket" "main" {
  bucket = "${var.s3_bucket_name}"
  acl    = "private"

  cors_rule {
    allowed_methods = "${var.s3_cors_allowed_methods}"
    allowed_origins = "${var.s3_cors_allowed_origins}"
  }
}
```

```hcl
resource "aws_s3_bucket" "main" {
  bucket = "${var.s3_bucket_name}"
  acl    = "private"

  cors_rule {
    allowed_methods = []
    allowed_origins = []
  }
}
```

### References

None

**Comment by anGie44** [comment]
Noting here that the API allows `allowed_origins` to be an empty string but `allowed_methods` must include atleast one of the allowed HTTP methods, excluding the empty string.

**Comment by anGie44** [comment]
Closed by #7547
